src/aspire/source/coordinates.py

```diff
@@ -0,0 +1,558 @@
+        """
+        All subclasses create mrc_paths and coord_paths lists and pass them to
+        this method.
+        """
+        for i in range(num_micrographs):
+            # read in all coordinates for the given mrc using subclass's
+            # method of reading the corresponding coord file
+            self.particles += [
+                (i, coord) for coord in self._coords_list_from_file(coord_paths[i])
+            ]
+
+    @abstractmethod
+    def _coords_list_from_file(self, coord_file):
+        """
+        Given a coordinate file, convert the coordinates into box format, i.e. a
+        list of the form [lower left x, lower left y, x size, y size].
+        Subclasses implement according to the details of the files they read.
+        """
+
+    @staticmethod
+    def _box_coord_from_center(center, particle_size):
+        """
+        Convert a list `[x,y]` representing a particle center
+        to a list
+        `[lower left x, lower left y, particle_size, particle_size]`
+        representing the box around the particle in box format.
+        :param center: a list of length two representing a center
+        :param particle_size: the size of the box around the particle
+        """
+        # subtract off floor(particle size/2) from center coords
+        r = floor(particle_size / 2)
+        x, y = center[:2]
+        return [
+            # centers may be represented as floats in STAR files
+            # chop off the non integer part to account for this
+            int(x) - r,
+            int(y) - r,
+            particle_size,
+            particle_size,
+        ]
+
+    @staticmethod
+    def _center_from_box_coord(box_coord):
+        """
+        Convert a list
+        `[lower left x, lower left y, particle_size, particle_size]`
+        representing a particle in the box format to a list
+        `[x, y]` representing the particle center.
+        :param box_coord: a list of length 4 representing the particle box
+        """
+        # Get lower left corner x and y coordinates
+        # and particle size from the box coordinate
+        llx, lly, particle_size = box_coord[:3]
+        # turn lower left corner coordinate into center coordinate
+        r = floor(particle_size / 2)
+        return [llx + r, lly + r]
+
+    def _coords_list_from_star(self, star_file):
+        """
+        Given a Relion STAR coordinate file (generally containing particle centers)
+        return a list of coordinates in box format.
+        :param star_file: A path to a STAR file containing particle centers
+        """
+        df = StarFile(star_file).get_block_by_index(0).astype(float)
+        coords = list(zip(df["_rlnCoordinateX"], df["_rlnCoordinateY"]))
+        return [
+            self._box_coord_from_center(coord, self.particle_size) for coord in coords
+        ]
+
+    def _check_and_get_paths(self, files):
+        """
+        Turns all paths into absolute paths.
```

Comment:
In `RelionSource` we do this, but that might be because STAR files almost always have paths relative to a directory other than the CWD. 

My thought was in the case of unexpected behavior (if a `CoordinateSource` is instantiated in a script that's running somewhere other than the CWD, maybe), it's good to give the full filepaths when there's a problem. Also, if we may expect a mix of absolute and relative paths, it's less of a headache if they're all converted to absolute paths.

Reply:
Eh I don't know. We don't do this anywhere else, so I don't think we should start doing it here unless there's a good reason. I think users will be familiar enough with how CWD works that this shouldn't be an issue while debugging.

Reply:
Was able to remove that whole method so that we just unpack the `files` argument in the `__init__`

Reply:
Got it. Cool.